### PR TITLE
Yank gateway out to a separate repo

### PR DESCRIPTION
**Pull request by rht**

### CODE REVIEW

**Comment by rht** [comment]
`gx deduplicate` would be useful

**Comment by rht** [comment]
@whyrusleeping looks like it already does (the deduplication) in the latest version, by not adding the "hardcoded" vendor itself. Will update once it works.

File: vendor/QmXvNwFNR7CQJ1J5oJnjS4oAFv2ai6v9JZBkTBbJp6wRQf/go-ipfs-gateway/gateway_handler.go [hunk 1]
```diff
@@ -21,6 +21,8 @@ import (
 	path "github.com/ipfs/go-ipfs/path"
 	"github.com/ipfs/go-ipfs/routing"
 	uio "github.com/ipfs/go-ipfs/unixfs/io"
+
+	logging "github.com/rht/go-ipfs-gateway/vendor/QmQg1J6vikuXF9oDvm4wpdeAUvvkVEKW1EYDw9HhTMnP2b/go-log"
```
Comment (rht) [inline vendor/QmXvNwFNR7CQJ1J5oJnjS4oAFv2ai6v9JZBkTBbJp6wRQf/go-ipfs-gateway/gateway_handler.go]: Fix this namespace once "gx install <hash>" works.

**Comment by rht** [comment]
`gx install` stalls, most likely because the hashes it needs to refer to don't exist / pinned?

**Comment by rht** [comment]
cmd/ipfswatch and cmd/seccat can be readily yanked out as well.

File: vendor/QmXvNwFNR7CQJ1J5oJnjS4oAFv2ai6v9JZBkTBbJp6wRQf/go-ipfs-gateway/gateway_handler.go (hunk 1 above)
Comment (whyrusleeping) [inline vendor/QmXvNwFNR7CQJ1J5oJnjS4oAFv2ai6v9JZBkTBbJp6wRQf/go-ipfs-gateway/gateway_handler.go]: gx install works, but we need to decide on a final pathing format for gx before i'm comfortable pushing it all the way into go-ipfs. 

ref: https://github.com/whyrusleeping/gx/issues/21

Comment (rht) [inline vendor/QmXvNwFNR7CQJ1J5oJnjS4oAFv2ai6v9JZBkTBbJp6wRQf/go-ipfs-gateway/gateway_handler.go]: Updating the pathing of 3 packages won't cause serious merge conflicts, however.

**Comment by rht** [comment]
- [x] I have updated the dir-index-html hash to https://ipfs.io/QmaywEmHiYfhcv1gj8gTNq6jwcN7Cs1EJkkE9kksT1255V
- [ ] I can't `ipfs get` the go-log hash however, and when I tried to `gx publish` github.com/ipfs/go-log, I get  the hash https://ipfs.io/QmaywEmHiYfhcv1gj8gTNq6jwcN7Cs1EJkkE9kksT1255V instead.

**Comment by whyrusleeping** [comment]
>  I can't ipfs get the go-log hash however, and when I tried to gx publish github.com/ipfs/go-log, I get the hash https://ipfs.io/QmaywEmHiYfhcv1gj8gTNq6jwcN7Cs1EJkkE9kksT1255V instead.

This is because paths in the installed package have been rewritten for godep compatibility. This will also be an issue for gx, unless we explicitly disallow sub-packages. (as the subpackages will need the hash of the package to have correct imports, this all is done in a `post-install` step)

**Comment by whyrusleeping** [comment]
once #2256 lands, we can redo this using gx

**Pull request closed without merging**